src/routers/UpdateMatchAvailabilitiesRouter.ts

```diff
@@ -0,0 +1,80 @@
+import { Request } from 'express';
+import Constants from '../common/constants';
+import Availability from '../entities/Availability';
+import AvailabilityRepo from '../repos/AvailabilityRepo';
+import MatchRepo from '../repos/MatchRepo';
+import AuthenticatedApplicationRouter from '../utils/AuthenticatedApplicationRouter';
+
+class UpdateMatchAvailabilitiesRouter extends AuthenticatedApplicationRouter<void> {
+  constructor() {
+    super('POST');
+  }
+
+  getPath(): string {
+    return '/availabilities/';
+  }
+
+  async content(req: Request): Promise<void> {
+    const { matchID, schedule } = req.body;
+
+    if (!matchID) throw Error('Missing required matchID');
+    if (!schedule) throw Error('Missing required schedule list');
+
+    // Logic to update match status
+    const match = await MatchRepo.getMatchByID(matchID);
+    let status = Constants.MATCH_CREATED;
+    if (match.status !== Constants.MATCH_CREATED && match.status !== Constants.MATCH_PROPOSED) {
+      throw Error('Cannot update match availability at this stage!');
+    }
+    if (match.status === Constants.MATCH_CREATED) {
+      status = Constants.MATCH_PROPOSED;
+    }
+    if (match.status === Constants.MATCH_PROPOSED) {
+      status = Constants.MATCH_ACTIVE;
+      if (schedule.length !== 1 || schedule[0].times.length !== 1) {
+        throw Error('Schedule must contain exactly one day and one time to finalize the match!');
+      }
+      // TODO: set meeting time properly
+    }
+
+    schedule.forEach((availability) => {
+      const { day, times } = availability;
+
+      if (day) {
```

Comment:
would probably be good to throw this into some utils since we do this kind of checking in two places

Reply:
oh wait everything below looks like code from UpdateAvailabilities, not sure if you made a minor tweak somewhere for this case but would be good to factor out any common code if possible